gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,127 @@
+
+```json
+{
+    "commit-hash": "XXXXX",
+    "config": {
+        "ipfs": "<IPFS_NODE_URL>",
+        "ipfs-concurrency": "4",
+        "ipfs-timeout": "10000",
+        "min-signal": "100",
+        "period": "300",
+        "grace_period": "0",
+        "supported-networks": "mainnet,gnosis,xdai,eip155:1,eip155:100,avalanche,eip155:43114,arbitrum-one,eip155:42161,celo,eip155:42220,fantom,eip155:250,matic,eip155:137",
+        "supported_data_source_kinds": "ethereum,ethereum/contract,file/ipfs,substreams,file/arweave",
+        "subgraph": "https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-arbitrum",
+        "subgraph_availability_manager_contract": "CONTRACT_ADDRESS",
+        "oracle_index": "ORACLE_INDEX",
+        "url": "<RPC_URL>"
+    }
+}
+
+```
+
+It is strongly recommended that Oracle operators diversify their sources for `IPFS_NODE_URL` and `RPC_URL` configurations. Utilizing varied sources for these critical components not only enhances the robustness and reliability of the network but also mitigates potential risks associated with single points of failure.
+
+## 5. Testnet
+
+In addition to their operations on Arbitrum One, all oracle operators are required to run a corresponding version on the Arbitrum Sepolia testnet. This setup is necessary to create a testing environment that closely mirrors the production environment, facilitating thorough testing and verification before deployment to production. Part of this requirement involves sending a transaction with the commit hash and configuration to the `DataEdge` smart contract on the testnet, following the format mentioned in the section above.
+
+## 6. Security
+
+Oracle operators are expected to follow best practices in security to protect their systems from unauthorized access and potential vulnerabilities.
+
+## 7. Conflict Resolution and Recourse
+
+Given that all oracle operators are mandated to run the same software version, uniformity in voting outcomes is expected. Discrepancies in oracle votes are therefore an area of concern and warrant investigation. In instances where voting differences occur or concerns are raised to the foundation or the core developers, the Technical Advisory Board and the Subgraph Availability Oracles operators will appoint an investigator to look into the issue. This investigation aims to determine whether the discrepancy was due to an honest failure or a result of malicious intent. The investigators decision will be final.
+
+If an oracle operator is found to be acting maliciously, the Graph Council will nominate a new operator to replace them. In scenarios where an oracle consistently deviates from the expected voting pattern due to honest errors, The Graph Council will evaluate its performance. Persistent non-compliance with the charter's guidelines, even if unintentional, may lead to the replacement of the oracle operator to maintain the integrity and reliability of the network.
```

Comment:
Should there be any contingency plans outlined for coordinated malicious behavior from 3 or more operators, of if 3 operators experienced the same honest mistake and a transaction has gone through?

Reply:
With 5 operators in our system, it can still function correctly even if up to 2 operators are malicious or make honest mistakes, allowing time for corrective actions, such as replacing operators. If a transaction is processed due to an honest mistake, operators must rectify the error, and the oracles will automatically correct the transaction.

Reply:
thanks for clarifying! is this something you would include in the GIP, such as how do we expect the operators to realize there was an erroneous transaction, and the process to rectify the error?